feat(EVS): add resource to operate evs snapshot group (#7389)
huaweicloud/terraform-provider-huaweicloud: docs/resources/evs_snapshot_group.md

---
subcategory: "Elastic Volume Service (EVS)"
layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_evs_snapshot_group"
description: |-
  Manages an EVS snapshot group resource within HuaweiCloud.
---

# huaweicloud_evs_snapshot_group

Manages an EVS snapshot group resource within HuaweiCloud.

> -> The snapshot group action only can be created while no snapshot creating.

Reviewer: ```suggestion
-> The snapshot group only can be created while no snapshot creating.
```

## Example Usage

```hcl
variable "region" {}
variable "name" {}
variable "description" {}
variable "volume_ids" {}
variable "tags" {}
variable "enterprise_project_id" {}
variable "server_id" {}
variable "instant_access" {}
variable "incremental" {}

resource "huaweicloud_evs_snapshot_group" "example" {
  region                = var.region
  name                  = var.name
  description           = var.description
  volume_ids            = var.volume_ids
  tags                  = var.tags
  enterprise_project_id = var.enterprise_project_id
  server_id             = var.server_id
  instant_access        = var.instant_access
  incremental           = var.incremental
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource. If omitted, the
  provider-level region will be used. Changing this creates a new resource.

* `server_id` - (Optional, String, NonUpdatable) Specifies the server ID to which the snapshot group are attached.

>   Must specified one of server_id or volume_ids.

Reviewer: ```suggestion
  Must specified one of `server_id` or `volume_ids`.
```

  Must specified one of server_id or volume_ids.

* `instant_access` - (Optional, Bool, NonUpdatable) Specifies whether to enable instant access for the snapshot group.
  Possible values are **true** (enable) and **false** (disable). Default is **false**.

* `enterprise_project_id` - (Optional, String, NonUpdatable) Specifies the enterprise project ID for the snapshot
  group.

* `incremental` - (Optional, Bool, NonUpdatable) Specifies whether to create an incremental snapshot.
  Default is **false**.

* `tags` - (Optional, Map) Specifies the key/value pairs to be associated with the snapshot group.

* `name` - (Optional, String) Specifies the snapshot group name. The maximum length is `255` bytes.

* `description` - (Optional, String) Specifies the snapshot group description. The maximum length is `255` bytes.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The resource ID.

* `created_at` - The time when the snapshot group was created.

* `status` - The snapshot group status.

* `updated_at` - The time when the snapshot group was updated.

> * `name` - The snapshot group name.

Reviewer: ```suggestion
```

> * `description` - The snapshot group description.

Reviewer: ```suggestion
```

> * `enterprise_project_id` - The ID of the enterprise project associated with the snapshot group.

Reviewer: ```suggestion
```

> * `tags` - The tags of the snapshot group.

Reviewer: ```suggestion
```

> * `server_id` - The server ID to which the snapshot group are attached.

Reviewer: ```suggestion
```

variable "region" {}
variable "name" {}
variable "description" {}
variable "volume_ids" {}
variable "tags" {}
variable "enterprise_project_id" {}
variable "server_id" {}
variable "instant_access" {}
variable "incremental" {}

resource "huaweicloud_evs_snapshot_group" "example" {
  region                = var.region
  name                  = var.name
  description           = var.description
  volume_ids            = var.volume_ids
  tags                  = var.tags
  enterprise_project_id = var.enterprise_project_id
  server_id             = var.server_id
  instant_access        = var.instant_access
  incremental           = var.incremental
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource. If omitted, the
  provider-level region will be used. Changing this creates a new resource.

* `server_id` - (Optional, String, NonUpdatable) Specifies the server ID to which the snapshot group are attached.
  Must specified one of `server_id` or `volume_ids`.

* `volume_ids` - (Optional, List, NonUpdatable) Specifies the list of volume IDs to be included in the snapshot group.

>   The value is in the ids=['id1','id2',...,'idx'] format. If server_id is set, this parameter can only be used to set

Reviewer: ```suggestion
  The value is in the **[id1, id2, ..., idx]** format. If `server_id` is set, this parameter can only be used to set
```

---
subcategory: "Elastic Volume Service (EVS)"
layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_evs_snapshot_group"
description: |-

>   Manages a resource to create, update, show, and delete EVS snapshot group within HuaweiCloud.

Reviewer: ```suggestion
  Manages an EVS snapshot group resource within HuaweiCloud.
```

---

# huaweicloud_evs_snapshot_group

Manages a resource to create, update, show, and delete EVS snapshot group within HuaweiCloud.

> -> The current resource is a one-time resource, and destroying this resource will not recover the deleted snapshot

Reviewer: This sentence is unnecessary

Reviewer: It is only necessary to specify the prerequisites for using this resource

  Must specified one of server_id or volume_ids.

* `volume_ids` - (Optional, List, NonUpdatable) Specifies the list of volume IDs to be included in the snapshot group.
  Must specified one of server_id or volume_ids.

* `instant_access` - (Optional, Bool, NonUpdatable) Specifies whether to enable instant access for the snapshot group.
  Possible values are **true** (enable) and **false** (disable). Default is **false**.

* `enterprise_project_id` - (Optional, String, NonUpdatable) Specifies the enterprise project ID for the snapshot
  group.

* `incremental` - (Optional, Bool, NonUpdatable) Specifies whether to create an incremental snapshot.
  Default is **false**.

* `tags` - (Optional, Map) Specifies the key/value pairs to be associated with the snapshot group.

* `name` - (Optional, String) Specifies the snapshot group name. The maximum length is `255` bytes.

* `description` - (Optional, String) Specifies the snapshot group description. The maximum length is `255` bytes.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The resource ID.

> * `job_id` - The task ID.

Reviewer: The `job_id` does not need to be reflected in the resource

> * `snapshot_group_id` - The snapshot group ID.

Reviewer: No need, just an `id` is enough
